test/protocols/binary_test.exs

```diff
@@ -1,80 +1,183 @@
 defmodule BinaryProtocolTest do
+  use ThriftTestCase, gen_erl: true
   use ExUnit.Case
 
-  @thrift_file_path "./test/fixtures/app/thrift/simple.thrift"
+  @moduletag :integration
 
-  import ParserUtils
+  def round_trip_struct(data, serializer_mf, deserializer_mf) do
+    {serializer_mod, serializer_fn} = serializer_mf
+    {deserializer_mod, deserializer_fn} = deserializer_mf
 
-  setup_all do
-    @thrift_file_path
-    |> parse_thrift
-    |> compile_module
+    serialized = :erlang.apply(serializer_mod, serializer_fn, [:struct, data])
```

Comment:
Why does it require that first :struct parameter? Is it an artifact of the implementation? If so, I think we should have serialize/1 as the exposed API people code against and defp the rest. I.e.:
```
def serialize(struct) do
  serialize(:struct, struct)
end
defp serialize(:struct, struct) do
  # ...
end
```